### PR TITLE
Convert the acceleration to the correct SI units; add a delay to fix SPI writes.

### PR DESCRIPTION
**Pull request by Claegtun**
### Brief

- Converts the accelerometer to ms^-2 instead of g to match the main codebase.
- Adds a 10-ms delay after setting the IMU's clock-source lest it ignores writing to the registers.

### Problem

For whatever reason unbeknown to me, the IMU registers will not be written to unless there is a delay after setting the clock-source in the `PWR_MGMT_1` register. 

Originally, this problem manifested itself where the accelerometer values were half of what were expected. Reading back the `ACCEL_CONFIG` register, it was revealed that it was still at the reset value, i.e. set to ±2 g, instead of the chosen ±4 g.

I found out that a delay was needed when I was debugging with breakpoints in `IMU::init`. If I had a breakpoint before the `ACCEL_CONFIG2` register was written to, then the register would have the correct value, and the accelerometer would have the expected values.

### Solution

I pin-pointed the spot at which the delay needed to be. It needed to be after the second writing of the `PWR_MGMT_1` register. Initially, a delay of 1 ms seemed to work. So, I made it an order of magnitude longer to be sure, i.e. 10 ms.

I cannot find anywhere in the datasheet that says that this is needed. The only specification like it is that the startup time for the register read/write ([page-12](https://invensense.tdk.com/wp-content/uploads/2021/03/DS-000143-ICM-20689-TYP-v1.1.pdf)) needs to be at least 5 ms from sleep and 100 ms from power-up. However, there is already a 1-s delay before `IMU::init`. Otherwise, it may be that the soft-reset at the very start of `IMU::init` counts as 'sleep'. If so, then that may be the reason.

Any answers and theories as to why are welcome.

### CODE REVIEW

**Comment by JohanneMontano** [comment]
What was the state of the values for the temp? Were they doubled as well?

**Comment by JohanneMontano** [comment]
there might be a rogue LSL that's hiding in here somewhere

**Comment by Claegtun** [comment]
I have recently found out something else that might be the issue. Ignore this draft PR for now.

File: NUSense/Core/Src/imu.cpp [hunk 1]
```diff
@@ -186,9 +188,9 @@ namespace nusense {
                                                    .y = static_cast<int16_t>(gyro.y.l | (gyro.y.h << 8)),
                                                    .z = static_cast<int16_t>(gyro.z.l | (gyro.z.h << 8))}};
         // convert from big to little endian and then scale
```
Comment (ysims) [inline NUSense/Core/Src/imu.cpp:190]: might be good to comment on the conversion to SI units and what they are, eg acc to metres per second and gyro to radians per second

Comment (Claegtun) [reply NUSense/Core/Src/imu.cpp]: Done.

**Review by TrentHouliston — APPROVED** [review]

**Pull request merged**